src/dune_rules/lib.ml

```diff
@@ -947,9 +948,15 @@ end = struct
           let open Resolve.Memo.O in
           let* vlib = resolve_forbid_ignore name in
           let virtual_ = Lib_info.virtual_ vlib.info in
-          match virtual_ with
-          | false -> Error.not_virtual_lib ~loc ~impl:info ~not_vlib:vlib.info
-          | true -> Resolve.Memo.return vlib
+          let parameterized = Lib_info.is_parameter vlib.info in
+          match virtual_, parameterized with
```

Comment:
nitpick: I read the variable name `parameterized` as "requires parameters" instead of "is a parameter"